### DISCUSSION TITLE
Mock patching during ChannelsLiveServerTestCases

### DISCUSSION THREAD

**Discussion by ckcollab** [django/channels#1738] [Q&A]
Thank you so much for channels, as always!

We're trying to write some selenium E2E tests that require django views to be mocked, and we're discovering that this doesn't seem very simple.. are there some best practices/tips for accomplishing mock patching Django views within [ChannelsLiveServerTestCase](https://channels.readthedocs.io/en/latest/topics/testing.html#channelsliveservertestcase)? 

Will gladly submit a PR to the docs with some examples, once we figure out a neat way to do it.

Relates to: https://github.com/django/channels/issues/707

**Comment by carltongibson** [comment]
Hi @ckcollab -- sorry no not really. You'll need to experiment and see what works. 

I think the comments on the linked issue are probably the best starting point. 

If you make progress. It'll be interesting to see what you come up with. 

I'm going to concert this to a discussion as I'm not sure it's actionable in its current state.

**Comment by claudep** [comment]
It was not in a channel context, but for the records, I was able to do some mocking by code like this (my use case was to avoid some requests calls triggered by the UI under tests):
```
from django.contrib.staticfiles.testing import StaticLiveServerTestCase
from django.test.testcases import LiveServerThread

class CustomLiveServerThread(LiveServerThread):
    def run(self):
        with mock.patch('some.patching.point', ...):
            super().run()

class MyLiveTests(StaticLiveServerTestCase):
    server_thread_class = CustomLiveServerThread
```
